File: IT/Oggetti/Esercizi/Java/Esercizio5.java
/* file Esercizio5.java */

class Spaceman {
	private String name;

	public Spaceman(String name) {
		setName(name);
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	@Override
	public String toString() {
		return "Sono l'astronauta " + this.name + "!";
	}
}

public class Esercizio5 {
	public static void main(String [] args) {
		Spaceman neil = new Spaceman("Neil Armstrong");
		System.out.println(neil);
	}
}

File: IT/Oggetti/Esercizi/C++/esercizio5.cpp
/* file esercizio5.cpp */

#include <iostream>

using std::cout;
using std::endl;
using std::string;
using std::ostream;

class Spaceman {
	private:
		string name;

	public:
		Spaceman(string);
		string getName();
		void setName(string);

	friend ostream& operator<<(ostream&, const Spaceman&);
};

Spaceman::Spaceman(string name)
{
	this->setName(name);
}

string Spaceman::getName() {
	return this->name;
}

void Spaceman::setName(string name) {
	this->name = name;
}

ostream& operator<<(ostream& os, const Spaceman& s) {
	os << "Sono l'astronauta " << s.name << "!";
	return os;
}

int main()
{
	Spaceman neil("Neil Armstrong");
	cout << neil << endl;
}

